src/utils/getTokenList.ts

```diff
@@ -89,6 +90,40 @@ export default async function getTokenList(
   throw new Error('Unrecognized list URL protocol.')
 }
 
+// loop to fetch all whitelist token
+export async function getTokenListV2(
+  listUrl: string,
+  resolveENSContentHash?: (ensName: string) => Promise<string>,
+): Promise<TokenList> {
+  return new Promise(async (resolve, reject) => {
+    let tokens: any[] = []
+    try {
+      const pageSize = 100
+      const maximumPage = 15
+      let page = 1
+      while (true) {
+        const { data } = await axios.get(`${listUrl}&pageSize=${pageSize}&page=${page}`)
+        page++
+        const tokensResponse = data.data.tokens ?? []
+        tokens = tokens.concat(tokensResponse)
+        if (tokensResponse.length < pageSize || page >= maximumPage) break // out of tokens, and prevent infinity loop
+      }
+    } catch (error) {
+      return reject(`Failed to download list ${listUrl}`)
+    }
+    const parsedData: TokenList = {
+      tokens,
+      name: 'KyberSwap Token List',
+      logoURI: 'https://kyberswap.com/favicon.png',
+      keywords: ['kyberswap', 'dmmexchange'],
+      version: { major: 0, minor: 0, patch: 0 },
+      timestamp: Date.now() + '',
+    }
+    formatTokensAddress(parsedData)
+    resolve(parsedData)
+  })
+}
```

Comment:
Is this gonna load all of tokens on started web instead of load more on demand?

Reply:
I dont think this approach is correct. Instead I suggest using [generator function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function*), which we can ask `getTokenListV2.next()` to `yield` more data.

Reply:
- load more only applies in the search modal. I fixed it.
- we will load all tokens with whitelist tokens in order to not break current logic of useAllToken(). 
- yes, generator is an alternative one, why this is incorrect